### PR TITLE
new pr for Turkish NER

### PR DESCRIPTION
**Pull request by merveenoyan**

### CODE REVIEW

**Comment by lhoestq** [comment]
Can you run `make style` to fix the code format ?

Also it looks like the file `file_downloaded/TWNERTC_TC_Coarse Grained NER_DomainIndependent_NoiseReduction.zip/TWNERTC_TC_Coarse Grained NER_DomainIndependent_NoiseReduction.DUMP` is missing inside the dummy_data.zip


(note that `TWNERTC_TC_Coarse Grained NER_DomainIndependent_NoiseReduction.zip` is a directory name, not an actual zip file)

**Review by lhoestq — COMMENTED** [review]
Looking good so far !

Can you format the code and fix the dummy data ? This way it will fix the fix and make it all green :)

After that feel free to add the datasets tags (the yaml part at the top of the dataset card)

**Comment by merveenoyan** [comment]
Hi Quentin, thank you for your patience with me. I've fixed the preprocessing pipeline, got this very weird error that Yacine told me to push. I've pushed it and after I'll find out that it will work, I will have my final pr on styling.

**Comment by lhoestq** [comment]
looks like you removed the dataset script file in your latest commit, is it expected ?

File: datasets/turkish_ner/TWNERTC_TC_Coarse Grained NER_DomainIndependent_NoiseReduction.DUMP [hunk 1]
```diff
@@ -0,0 +1,5 @@
+people	O O B-MISC O O B-LOCATION O B-PERSON O O O O O B-MISC O O O O O	24 yaşındaki 1.86 metre boyundaki Senegalli golcü Dabo , daha önce Ugs , Sion , Servette , St .
```
Comment (lhoestq) [inline datasets/turkish_ner/TWNERTC_TC_Coarse Grained NER_DomainIndependent_NoiseReduction.DUMP:1]: can you also remove this file ? :)

File: datasets/turkish_ner/README.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+language_creators:
+- expert-generated
+languages:
+- tr
+licenses:
+- cc-by-4.0
+multilinguality:
+- monolingual
+size_categories:
+- n>700K
```
Comment (lhoestq) [inline datasets/turkish_ner/README.md:13]:
```suggestion
- 100K<n<1M
```

File: datasets/turkish_ner/turkish_ner.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        ]
+
+    def _generate_examples(self, filepath, split):
+        """ Yields examples. """
+        logging.info("⏳ Generating examples from = %s", filepath)
+
+        with open(filepath, encoding="utf-8") as f:
+            id_ = -1
+            for id_, line in enumerate(f):
+                if line.startswith("-DOCSTART-") or line == "" or line == "\n":
```
Comment (lhoestq) [inline datasets/turkish_ner/turkish_ner.py:160]:
```suggestion
            for line in f:
                if line == "" or line == "\n":
```

**Review by lhoestq — APPROVED** [review]
Nice ! Thank you for adding this one :)

**Pull request merged**